DEVELOPER.md

```diff
@@ -17,7 +17,7 @@ There are two ways to upload Twinkle scripts to Wikipedia or another destination
 
 After the files are synced, ensure that [MediaWiki:Gadgets-definition][] contains the following lines:
 
-    * Twinkle[ResourceLoader|dependencies=ext.gadget.morebits,ext.gadget.select2,mediawiki.api,mediawiki.language|rights=autoconfirmed|type=general|peers=Twinkle-pagestyles]|Twinkle.js|twinkleprod.js|twinkleimage.js|twinklebatchundelete.js|twinklewarn.js|twinklespeedy.js|friendlyshared.js|twinklediff.js|twinkleunlink.js|friendlytag.js|twinkledeprod.js|friendlywelcome.js|twinklexfd.js|twinklebatchdelete.js|twinklebatchprotect.js|twinkleconfig.js|twinklefluff.js|twinkleprotect.js|twinklearv.js|twinkleblock.js|friendlytalkback.js|Twinkle.css
+    * Twinkle[ResourceLoader|dependencies=ext.gadget.morebits,ext.gadget.select2,mediawiki.api,mediawiki.language|rights=autoconfirmed|type=general|peers=Twinkle-pagestyles]|Twinkle.js|Twinkle.css|twinklearv.js|twinklewarn.js|twinkleblock.js|friendlywelcome.js|friendlyshared.js|friendlytalkback.js|twinklespeedy.js|twinkleprod.js|twinklexfd.js|twinkleimage.js|twinkleprotect.js|friendlytag.js|twinklediff.js|twinkleunlink.js|twinklefluff.js|twinkledeprod.js|twinklebatchdelete.js|twinklebatchprotect.js|twinklebatchundelete.js|twinkleconfig.js
     * morebits[ResourceLoader|dependencies=mediawiki.user,mediawiki.util,mediawiki.Title,jquery.ui,jquery.tipsy|hidden]|morebits.js|morebits.css
```

Comment:
Is there a reason the order of these is changed? It makes for a confusing diff on version control and when I compare it on https://www.diffchecker.com/e0QzkyMf I see no reason why the order of modules should be changed.

Reply:
@Huji See the note in the top message:

>The order in which the menu buttons appear is still within our control through MediaWiki:Gadgets-definition - the files are executed in the order in which they appear in the gadget definition line. The gadget definition on-wiki will need to be tweaked as done in DEVELOPER.md file.

This PR changes how and when the files are loaded.  Currently, it is `twinkle.js` which loads each module, via `Twinkle.load`.  Rather than hard-coding the order of modules in `twinkle.js`, this leverages the gadget definition to establish the order.  The difference is mostly for testing Twinkle.js itself, but it does make sense overall.  After this PR, it probably makes alternative installations easier to maintain.

Reply:
That being said, I think there's probably an argument to having `twinkleconfig` ahead of the others, although it shouldn't matter?  `fluff` could be moved to the very end.

Reply:
Yeah, that is what I mean. Don't move `fluff` or `welcome` or `xfd` or `arv` , etc. Only move things that need to be moved. Or am I missing the point?

Reply:
IIRC I copied the sequence from the original code (which was based on sound logic) to the gadget def (which was a random jumble). 

In any case, it's clear from the diff that net lengths of the old and new strings are the same -- so nothing was added or removed. I am not sure what's confusing. 

> That being said, I think there's probably an argument to having twinkleconfig ahead of the others, although it shouldn't matter? fluff could be moved to the very end.

Isn't it the other way round? We want things to load/execute fast. Functional modules like fluff should go first, not things like twinkleconfig which is loaded only on one page that's rarely visited.

Reply:
Well, if nothing is supposed to be changed, then there should not be any difference in this line of code; yet there is.

Do you need to rebase? Do you need to first, in a separate patch, reorder module names to what you think is desirable (and explain why), and then use it here without change?

In any case, I'm going to stop pushing for this. I am not a primary maintainer of Twinkle, so it is not my place to push any harder. My entire point is this: each commit should ONLY change what is needed for that commit; it should not include any other changes.

Reply:
@Huji This matches what was in the `twinkleModules` list in twinkle.js, see https://github.com/azatoth/twinkle/pull/946/files#diff-551c5d596f19b4df338bbf4ab50919baL450-L459, which is how the loading order is currently determined.  This PR supersedes and does away with that list in favor of using the order in the gadget, so it's a one-time move to establish the proper loading order via the gadget.  It should work fine without changing the order, but the order of the items in the menu/tabs will be different.

@siddharthvp I meant more from a "base things you can't/shouldn't turn off" but good point!  Especially on fluff.

Reply:
@Amorymeltzer oh, now I get it. In that case, I think a new order is justified.